### PR TITLE
Implement create project endpoint

### PR DESCRIPTION
**Pull request by Aratz**
**What problems does this PR solve?**
During a delivery workflow, we need to create a project in DDS. This PR adds an endpoint in the API to create such a project, as well as a new table in the database to keep track of the correspondence between DDS project ids and projects in Clarity.

**How has the changes been tested?**
Unit and integration tests have been added. Creating a project in DDS have been successfully tested locally. Testing in staging with the last DDS release will be performed after a new version of arteria-delivery is released.

**Reasons for careful code review**
If any of the boxes below are checked, extra careful code review should be initiated.

  - [ ] This PR contains code that could remove data

### CODE REVIEW

**Review by matrulda — APPROVED** [review]
Great work! :clap: Looks good to me.

One general thing I thought of that isn't directly tied to this PR, but anyway: 
I wonder if we should add dds-cli to the requirements? We could also use the DDS version that is installed in the uppmax module system or update the [dds-cli role](https://github.com/NationalGenomicsInfrastructure/miarka-provision/tree/main/roles/dds_cli) so that the client also is installed in the arteria-delivery env. Seems most convenient to go with the first. The only possible issue I can think of is that BIO will use the version installed in the NGI environment on Miarka and it could possibly not be the same version that the service will use. But I have a hard time seeing that it would be an issue to activate the token with one version of the client and and make propjects + deliver data with another one. 

While writing this I recall that we said the the dds_cli role would only stay for the testing period and we should use the module system in the long run. So, that would mean that we would need a `module load` step somewhere in the service, which would work when running it on a non-uppmax system. Leading me back to: it's probably a good idea to include it in the prod requirements. 

What do you think?

File: tests/unit_tests/services/test_dds.py [hunk 1]
```diff
@@ -159,3 +171,53 @@ def test_possible_to_delivery_by_staging_id_and_skip_mover(self):
...
+Project created with id: snpseq00003
+User forskare was associated with Project snpseq00003 as Owner=True. An e-mail notification has not been sent.
+Invitation sent to email@adress.com. The user should have a valid account to be added to a
+project"""
+
+        self.assertEqual(DDSService._parse_dds_project_id(dds_output), "snpseq00003")
+
+    @gen_test
+    def test_create_project(self):
+        self.mock_dds_project_repo.project_exists.return_value = False
```
[the remark this answers is not in the document]
Comment (Aratz) [reply tests/unit_tests/services/test_dds.py:186]: Ha, I think it's a leftover from a method I had when I mixed up delivery_project_ids and project_names, I'll take it away :+1:

**Comment by Aratz** [comment]
Good point :thinking:  If using different versions of dds could mess things up I would hope they would issue deprecation messages so I don't think this will be an issue for us. So I'd go with the first option :+1:

Does the service have to handle `module load` ? I was hopping this would go in the `sourceme.sh` file instead :confused:

**Comment by Aratz** [comment]
Also, when editing the st2 workflow I realized the service needs to return the DDS project id so that we know where to deliver the data in the next actions so I've fixed that too now.

**Comment by matrulda** [comment]
> Good point thinking If using different versions of dds could mess things up I would hope they would issue deprecation messages so I don't think this will be an issue for us. So I'd go with the first option +1
> 
> Does the service have to handle `module load` ? I was hopping this would go in the `sourceme.sh` file instead confused
We could have it in a source file for funk_004, but it could potentially interfere with other services then, so we only want it to happen in the session where the arteria-delivery service is started. 

If we add it to the requirements we won't have to think about it, because it will be installed in the conda env where the service is running. So, let's go for option 1! 

>Also, when editing the st2 workflow I realized the service needs to return the DDS project id so that we know where to deliver the data in the next actions so I've fixed that too now.
Ah didn't think of that, good catch! :+1: Will check your changes now.

**Comment by matrulda** [comment]
Ok, looks good! If you just add dds-cli to https://github.com/arteria-project/arteria-delivery/blob/master/requirements/prod I think we can merge this.

**Comment by Aratz** [comment]
Actually, I did that yesterday but forgot to push :smile: , so the commits are shown before your comments in the PR

**Comment by matrulda** [comment]
Ah sneaky! Coolio, then I'll merge.

**Pull request merged**